pkg/workloads/reservation.go

```diff
@@ -1211,6 +1226,76 @@ func (a *API) setWorkloadDelete(ctx context.Context, db *mongo.Database, w types
 	return w, errors.Wrap(types.WorkloadPush(ctx, db, w), "could not push workload to delete in queue")
 }
 
+func (a *API) checkAllowedPublicIPs(workload types.WorkloaderType, db *mongo.Database) error {
+	if workload.GetWorkloadType() != generated.WorkloadTypePublicIP {
+		return nil
+	}
+
+	ipWorkload := workload.Workloader.(*generated.PublicIP)
+
+	var nodeFilter directory.NodeFilter
+	nodeFilter = nodeFilter.WithNodeID(ipWorkload.NodeId)
+	node, err := nodeFilter.Get(context.Background(), db, false)
+	if err != nil {
+		return errors.Wrap(err, "failed to retrieve node id for ip")
+	}
+
+	var farmFilter directory.FarmFilter
+	farmFilter = farmFilter.WithID(schema.ID(node.FarmId))
+	farm, err := farmFilter.Get(context.Background(), db)
+	if err != nil {
+		return errors.Wrap(err, "failed to retrieve farm")
+	}
+
+	found := false
+	for _, farmIP := range farm.IPs {
```

Comment:
I think this entire loop can done in mongo with one call.
- If you search with an element in mongo the filter matches the full element so let's assume you have an array like this  [..., {ip: 1.1.1.1. reservation: 0}, ...] 
- if you do an update operation like `update({ip: 1.1.1.1, reservation: 0}, with {reservation: 100}) it will only work if the reservation is actually 0 (free ip) otherwise you will get an error that this did not match any value.
- If it's free it will get updated atomically and a success will be returned.

so no need for load farm, find ip check value, then update 

Same can be done for de-allocation